src/suit_directive.c

```diff
@@ -45,9 +45,19 @@ static int decode_encryption_info(struct zcbor_string enc_info_bstr, struct suit
 	enc_info->aad.value = suit_aad_aes256_gcm;
 	enc_info->aad.len = sizeof(suit_aad_aes256_gcm);
 
-	enc_info->kw_alg_id = suit_cose_aes256_kw;
-	enc_info->kw_key.aes.key_id = enc_info_cbor.COSE_Encrypt_recipients.COSE_recipients_protected_l_unprotected.rec_header_map_key_id;
-	enc_info->kw_key.aes.ciphertext = enc_info_cbor.COSE_Encrypt_recipients.COSE_recipients_protected_l_ciphertext;
+	enc_info->kw_alg_id = enc_info_cbor.COSE_Encrypt_recipients.COSE_recipients_protected_l_unprotected.rec_header_map_alg_id.supported_kdf_algs_choice;
+
+	switch (enc_info->kw_alg_id) {
+		case suit_cose_aes256_kw:
+			enc_info->kw_key.aes.key_id = enc_info_cbor.COSE_Encrypt_recipients.COSE_recipients_protected_l_unprotected.rec_header_map_key_id;
+			enc_info->kw_key.aes.ciphertext = enc_info_cbor.COSE_Encrypt_recipients.COSE_recipients_protected_l_ciphertext_bstr;
```

Comment:
Please check if ciphertext is not `nil`

Reply:
Did the validation in a bit different way, take a look now.